Oqtane.Client/Modules/Controls/LocalizableComponent.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using Microsoft.AspNetCore.Components;
+using Microsoft.Extensions.Localization;
+using Oqtane.Shared;
+
+namespace Oqtane.Modules.Controls
+{
+    public class LocalizableComponent : ModuleControlBase
+    {
+        private IStringLocalizer _localizer;
+
+        [Parameter]
+        public string ResourceKey { get; set; }
+
+        protected bool IsLocalizable { get; private set; }
+
+        protected string Localize(string name)
+        {
+            if (!IsLocalizable)
+            {
+                return null;
+            }
+
+            var key = $"{ResourceKey}.{name}";
+ 
+            return _localizer?[key] ?? key;
+        }
+
+        protected override void OnParametersSet()
+        {
+            if (!String.IsNullOrEmpty(ResourceKey))
+            {
+                if (ModuleState?.ModuleType != null)
+                {
+                    var moduleType = Type.GetType(ModuleState.ModuleType);
+                    var localizerTypeName = $"Microsoft.Extensions.Localization.IStringLocalizer`1[[{moduleType.AssemblyQualifiedName}]], Microsoft.Extensions.Localization.Abstractions";
+                    var localizerType = Type.GetType(localizerTypeName);
+
+                    // HACK: Use ServiceActivator instead of injecting IHttpContextAccessor, because HttpContext throws NRE in WebAssembly runtime
```

Comment:
Using `IHttpContextAccessor` in a Blazor Server app could expose you to security vulnerabilities. Components do not guarantee they execute in the context of a user's request (think a timer that's updated in the background or a chat component that causes other instances of chat components to update when you post a message). You should not use it in general since it's an anti-pattern, but definitely not in a server app.

Reply:
Thanks @pranavkm for pointing to this

Reply:
@hishamco are you going to be able to take a look at this? Or should I make an attempt at it?

Reply:
@sbwalker if you meant `IStringLocalizerFactory` I already started today on this on, but one point I'm still think about, the factory will create the `IStringLocalizer` but my aim in the PR was to retrieve it from the DI. I will update you on this soon, before releasing `2.0.0`

Reply:
Seems something broken in localization, I will try to figure out what happen into the latest few PRs